src/shared/bus-util.c

```diff
@@ -1593,3 +1594,103 @@ int bus_add_implementation(sd_bus *bus, const BusObjectImplementation *impl, voi
 
         return 0;
 }
+
+static bool non_fallback_implementation(const BusObjectImplementation *impl) {
+        for (const BusObjectVtablePair *p = impl->vtables; p->vtable; p++)
+                if (!p->object_find)
+                        return true;
+        return false;
+}
+
+static const BusObjectImplementation* find_implementation(
+                const char *pattern,
+                bool allow_fallback,
+                const BusObjectImplementation* const* bus_objects) {
+
+        for (unsigned i = 0; bus_objects[i]; i++) {
```

Comment:
size_t. we iterate through memory objects here... in theory there could be as many entries in the array as we have addressable memory...